packages/mui-lab/src/TreeItem/TreeItem.js

```diff
@@ -272,7 +272,11 @@ const TreeItem = React.forwardRef(function TreeItem(inProps, ref) {
   function handleFocus(event) {
     // DOM focus stays on the tree which manages focus with aria-activedescendant
     if (event.target === event.currentTarget) {
-      ownerDocument(event.target).getElementById(treeId).focus({ preventScroll: true });
+      if (typeof event.target.getRootNode === 'function') {
+        event.target.getRootNode().getElementById(treeId).focus({ preventScroll: true });
+      } else {
+        ownerDocument(event.target).getElementById(treeId).focus({ preventScroll: true });
+      }
```

Comment:
I think that `.getElementById(treeId).focus({ preventScroll: true });` should be deduplicated.

Reply:
Good point, I've updated the PR.